File: frontend/lib/services/auth_service.dart
import 'dart:convert';
import 'package:frontend/components/storages/auth_storage.dart';
import 'package:flutter_dotenv/flutter_dotenv.dart';
import 'package:http/http.dart' as http;

class AuthService {
  final String _baseUrl = '${dotenv.env['BASE_URL']}/auth';
  final AuthStorage _authStorage = AuthStorage();

  // Método para autenticar o usuário
  Future<bool> login({required String email, required String password}) async {
    try {
      final response = await http.post(
        Uri.parse(_baseUrl),
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode({
          'email': email,
          'password': password,
        }),
      );

      if (response.statusCode == 200) {
        final data = jsonDecode(response.body);
        String token = data['access_token'];

        // Salvar o token para uso futuro
        await _authStorage.saveToken(token);
        return true;
      } else {
        print('Falha no login: ${response.statusCode}');
        return false;
      }
    } catch (e) {
      print('Erro ao realizar login: $e');
      return false;
    }
  }

  // Método para verificar se o usuário está autenticado
  Future<bool> isLoggedIn() async {
    final token = await _authStorage.getToken();
    return token != null;
  }

  // Método para logout
  Future<void> logout() async {
    await _authStorage.removeToken();
  }
}

File: frontend/src/services/auth_service.ts
import { AuthStorage } from "../storage/auth_storage";
import decodeBase64 from "../utils/decode";
import httpClient from "./http_client";

export class AuthService {
  private endpoint = "/auth";

  // Método para autenticar o usuário
  async login(email: string, password: string): Promise<boolean> {
    try {
      const { data } = await httpClient.post(this.endpoint, {
        email,
        password,
      });

      const token = data.access_token;
      const emailDecoded = decodeBase64(data.userInfo.email);
      const nameDecoded = decodeBase64(data.userInfo.name);
      const uuid = data.userInfo.uuid;

      await AuthStorage.saveSession({
        token,
        email: emailDecoded,
        firstName: nameDecoded,
        uuid,
      });

      return true;
    } catch (error: any) {
      const status = error.response?.status || "N/A";
      console.warn(`Falha no login: ${status}`);
      return false;
    }
  }

  // Verifica se está autenticado
  async isLoggedIn(): Promise<boolean> {
    const session = await AuthStorage.getSession();
    return !!session?.token;
  }

  // Logout
  async logout(): Promise<void> {
    await AuthStorage.removeSession();
  }
}
